source/parser/comptime.civet

```diff
@@ -139,10 +139,31 @@ function serialize(value: ???): string
             when Array::
               `[${(val as ???[]).map(recurse).join ","}]`
             when Object::
-              "{" + (
-                for own key, value in val as {[key: string]: ???}
-                  `${JSON.stringify key}:${recurse value}`
-              ).join(",") + "}"
+              // The string representing the object's own enumerable, writable properties
+              objStr .= '{'
+              // The string representing other property descriptors, if applicable, not including the outermost braces
+              descStr .= ''
+              descriptors := Object.getOwnPropertyDescriptors(val) as Record<string | symbol, PropertyDescriptor>
+              for prop of (Object.getOwnPropertyNames(val) as (string|symbol)[]) ++ Object.getOwnPropertySymbols(val)
+                desc := descriptors[prop]
+                if desc.enumerable and desc.configurable and desc.writable
+                  if prop <? 'symbol'
+                    objStr += `[${recurse prop}]`
+                  else
+                    objStr += JSON.stringify prop
+                  objStr += `:${recurse desc.value},`
+                else
+                  if prop <? 'symbol'
+                    descStr += `[${recurse prop}]`
+                  else
+                    descStr += JSON.stringify prop
+                  descStr += `:${recurse desc},`
+              objStr += '}'
```

Comment:
Could we remove the trailing commas from `objStr` and `descStr`? Something like `if (objStr# > 1) objStr = objStr[...-1]` and `descStr = descStr[...-1]` inside the next `if`.